### PR TITLE
Create namespace in component

### LINKED ISSUE #2: Manage the default namespace
## Context

When installing this component, the default namespace `syn-prometheus-pushgateway` is not created since the component does not manage it. This would be a valuable feature.

Other components such as https://github.com/projectsyn/component-espejo do manage their namespace.

## Alternatives

Create the namespace manually:
```
kubectl create ns syn-prometheus-pushgateway
```

### PR DESCRIPTION
**Pull request by bastjan**
Create the namespace used for the component.

Fixes #2 

## Checklist

- [x] Keep pull requests small so they can be easily reviewed.

### CODE REVIEW

File: tests/golden/defaults/prometheus-pushgateway/prometheus-pushgateway/00_namespace.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  annotations: {}
+  labels:
+    name: syn-prometheus-pushgateway
```
Comment (ccremer) [inline tests/golden/defaults/prometheus-pushgateway/prometheus-pushgateway/00_namespace.yaml:6]: while we're at it, shouldn't it also contain `SYNMonitoring=main` or so? (or whatever the label's name, I haven't checked it)

Comment (bastjan) [reply tests/golden/defaults/prometheus-pushgateway/prometheus-pushgateway/00_namespace.yaml:6]: Not sure yet. The label is not enough for the rules to be picked up. At least on OpenShift 4.

**Review by glrf — APPROVED** [review]

File: tests/golden/defaults/prometheus-pushgateway/prometheus-pushgateway/00_namespace.yaml (hunk 1 above)
Comment (simu) [reply tests/golden/defaults/prometheus-pushgateway/prometheus-pushgateway/00_namespace.yaml:6]: OCP4 requires label `openshift.io/cluster-monitoring=true` to allow the OCP4 cluster monitoring to pick up the rules.

File: class/defaults.yml [hunk 2]
```diff
@@ -1,6 +1,7 @@
 parameters:
   prometheus_pushgateway:
     namespace: syn-prometheus-pushgateway
+    createNamespace: true
```
Comment (simu) [inline class/defaults.yml:4]: consider using `create_namespace` to keep in line with the other parameters (yes I know it's currently a mess across components).

```suggestion
    create_namespace: true
```

Also update the documentation to include this parameter and describe when/why it's necessary.

File: component/main.jsonnet [hunk 3]
```diff
@@ -6,6 +6,7 @@ local inv = kap.inventory();
 local params = inv.parameters.prometheus_pushgateway;
 
 {
+  [if params.createNamespace then '00_namespace']: kube.Namespace(params.namespace),
```
Comment (simu) [inline component/main.jsonnet:9]: Use the following to allow Prometheus to pick up the rules in the namespace:

```suggestion
local nsLabels = 
  if inv.parameters.facts.distribution == 'openshift4' then
    {
      'openshift.io/cluster-monitoring': 'true',
    }
  else
    {
      SYNMonitoring: 'main',
    };

{
  [if params.createNamespace then '00_namespace']:
    kube.Namespace(params.namespace) {
      labels+: nsLabels,
    },
```

**Review by simu — CHANGES_REQUESTED** [review]

File: tests/golden/defaults/prometheus-pushgateway/prometheus-pushgateway/00_namespace.yaml (hunk 1 above)
Comment (bastjan) [reply tests/golden/defaults/prometheus-pushgateway/prometheus-pushgateway/00_namespace.yaml:6]: > OCP4 requires label openshift.io/cluster-monitoring=true

Thanks! Really really hard to find in the docs...

File: docs/modules/ROOT/pages/references/parameters.adoc [hunk 4]
```diff
@@ -10,6 +10,16 @@ default:: `syn-prometheus-pushgateway`
...
 
+
+== `create_namespace`
+
+[horizontal]
+type:: boolean
+default:: `true`
+
+Whether to create the namespace given by `namespace`.
+Must be set to false if the Pushgateway is installed to a namespace managed by another component.
```
Comment (ccremer) [inline docs/modules/ROOT/pages/references/parameters.adoc:21]:
```suggestion
Must be set to `false` if the Pushgateway is installed to a namespace managed by another component.
```

**Review by ccremer — APPROVED** [review]

File: component/rbac.libsonnet [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    rules: [
+      {
+        apiGroups: [ '' ],
+        resources: [ 'services', 'endpoints', 'pods' ],
+        verbs: [ 'get', 'list', 'watch' ],
+      },
+    ],
+  };
+
+local ocp_metrics_rolebinding(name, namespace, metrics_role) =
```
Comment (bastjan) [inline component/rbac.libsonnet:19]: @simu Copied from the rook component. Do you think it is worth creating a library in the openshift-monitoring component?

Comment (simu) [reply component/rbac.libsonnet:19]: There's a bunch of things that should be library methods in `openshift-monitoring`, this is definitely a candidate.

**Review by simu — APPROVED** [review]

**Pull request merged**